ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/TableSuppressBuilderTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import io.confluent.ksql.execution.context.QueryContext;
+import io.confluent.ksql.execution.plan.ExecutionStep;
+import io.confluent.ksql.execution.plan.ExecutionStepPropertiesV1;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.KTableHolder;
+import io.confluent.ksql.execution.plan.KeySerdeFactory;
+import io.confluent.ksql.execution.plan.TableSuppress;
+import io.confluent.ksql.execution.windows.KsqlWindowExpression;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.logging.processing.ProcessingLogger;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.serde.RefinementInfo;
+import io.confluent.ksql.serde.SerdeOption;
+import io.confluent.ksql.util.KsqlConfig;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.processor.StateStore;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class TableSuppressBuilderTest {
+
+  @Mock
+  private KsqlQueryBuilder queryBuilder;
+  @Mock
+  private ExecutionStep<KTableHolder<Struct>> sourceStep;
+  @Mock
+  private KTable<Struct, GenericRow> sourceKTable;
+  @Mock
+  private KTable<Struct, GenericRow> preKTable;
+  @Mock
+  private KTable<Struct, GenericRow> suppressedKTable;
+  @Mock
+  private RefinementInfo refinementInfo;
+  @Mock
+  private Formats internalFormats;
+  @Mock
+  private KsqlWindowExpression windowExpression;
+  @Mock
+  private KeySerdeFactory<Struct> keySerdeFactory;
+  @Mock
+  private MaterializedFactory materializedFactory;
+  @Mock
+  private  PhysicalSchema physicalSchema;
+  @Mock
+  private  Serde<GenericRow> valueSerde;
+  @Mock
+  private  Serde<Struct> keySerde;
+  @Mock
+  private  Materialized<Object, GenericRow, StateStore> materialized;
+  @Mock
+  private KTableHolder<Struct> tableHolder;
+  @Mock
+  private KTableHolder<Struct> suppressedtable;
+
+  private final QueryContext queryContext = new QueryContext.Stacker()
+      .push("bar")
+      .getQueryContext();
+  private TableSuppress<Struct> tableSuppress;
+  private BiFunction<LogicalSchema, Set<SerdeOption>, PhysicalSchema> physicalSchemaFactory;
+  private TableSuppressBuilder builder;
+
+  @Rule
+  public final MockitoRule mockitoRule = MockitoJUnit.rule();
```

Comment:
just wondering, where did you pick this up? it's really cool - I think it's better than our standard `@RunWith(MockitoJUnitRunner.class)`

Reply:
This is what `TableGroupByBuilderTest` uses so I just followed that, why do you think it is better than our standard `@RunWith(MockitoJUnitRunner.class)`?

Reply:
With `@RunWith` you can only have one runner, which means if you want to run it with another runner (`Enclosed.class` for example) you can't use mocks with the `@Mock` annotation. That, and it doesn't use reflection to find the class which is just always been a pet peeve of mine.